src/containers/PatientDetails/PatientDocument/index.tsx

```diff
@@ -22,17 +27,85 @@ export interface PatientDocumentProps {
     launchContextParameters?: ParametersParameter[];
     questionnaireId?: string;
     encounterId?: string;
+    onSubmit?: (formData: QuestionnaireResponseFormData) => Promise<any>;
     onSuccess?: (resource: QuestionnaireResponseFormSaveResponse) => void;
-    autosave?: boolean;
+    onCancel?: () => void;
+    onQRFUpdate?: (questionnaireResponse: QuestionnaireResponse) => void;
+    autoSave?: boolean;
     qrDraftServiceType?: QuestionnaireResponseDraftService;
+    alertComponent?: React.ReactNode | (() => React.ReactNode);
 }
 
 export function PatientDocument(props: PatientDocumentProps) {
-    const { autosave, qrDraftServiceType = 'local' } = props;
+    const { autoSave = false, qrDraftServiceType = 'local', onCancel, onSubmit } = props;
+
+    const params = useParams<{ questionnaireId: string; encounterId?: string }>();
+
+    const { draftQuestionnaireResponseRD, draftInfoMessage, onQRFUpdate, deleteDraft, submitDraft } =
```

Comment:
rename onQRFUpdate only in useQuestionnaireResponseDraft to saveDraft/updateDraft